docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,567 @@
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: aws
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+<div class="tab-pane fade" id="gcp-tab-2" markdown="1">
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: gcp
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+<div class="tab-pane fade" id="azure-tab-2" markdown="1">
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: azure
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+<div class="tab-pane fade" id="alibaba-tab-2" markdown="1">
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: alibaba
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+</div>
+</div>
+
+After creating the `MySQLInstanceRequirement` Crossplane will provision a
+database instance on your provider of choice. Once provisioning is complete, you
+should see `READY: True` in the output when you run:
+```
+kubectl get mysqlinstancerequirement.database.example.org my-db
+```
+
+You should also see a `Secret` in the `default` namespace named `db-conn` that
+contains fields for `username`, `password`, and `endpoint`:
+```
+kubectl get secrets db-conn
+```
+
+## Consume Infrastructure
+
+Because connection secrets are written as a Kubernetes `Secret` they can easily
+be consumed by Kubernetes primitives. The most basic building block in
+Kubernetes is the `Pod`. Let's define a `Pod` that will show that we are able to
+connect to our newly provisioned database.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: see-db
+  namespace: default
+spec:
+  containers:
+  - name: see-db
+    image: postgres:9.6
+    command: ['psql']
+    args: ['-c', 'SELECT current_database();']
+    env:
+    - name: PGDATABASE
+      value: postgres
+    - name: PGHOST
+      valueFrom:
+        secretKeyRef:
+          name: db-conn
+          key: endpoint
+    - name: PGUSER
+      valueFrom:
+        secretKeyRef:
+          name: db-conn
+          key: username
+    - name: PGPASSWORD
+      valueFrom:
+        secretKeyRef:
+          name: db-conn
+          key: password
+```
+
+This `Pod` simply connects to a PostgreSQL database and prints its name, so you
+should see the following output after creating it if you run `kubectl logs
+see-db`:
+```
+ current_database 
```

Comment:
I don't see any output for a while when I follow AWS. Then the following after 2-3 min:
```
psql: could not connect to server: Connection refused
	Is the server running on host "default-my-db-zdcs5-g6bvd.csguarq5weag.us-east-1.rds.amazonaws.com" (54.160.126.136) and accepting
	TCP/IP connections on port 5432?
```
Could it be related to database being `MySQL` and client container image postgres? We don't mount a port and MySQL's default is 3306.

Reply:
I see the same behavior in Azure, too.

Reply:
Ah yes at some point I switched from Postgres to MySQL, will update to use a MySQL client

Reply:
See https://github.com/crossplane/crossplane/pull/1514/files#r426487230 for a MySQL compatible example.